trailblazer/services/user_verification_service/user_verification_service.py

```diff
@@ -0,0 +1,62 @@
+from typing import Mapping
+
+from google.auth import jwt
+import requests
+
+from trailblazer.services.user_verification_service.exc import (
+    GoogleCertsError,
+    UserTokenVerificationError,
+)
+from trailblazer.store.models import User
+from trailblazer.store.store import Store
+
+
+class UserVerificationService:
+    """Service to verify the user."""
+
+    def __init__(self, store: Store, oauth_client_id: str):
+        self.store: Store = store
+        self.oauth_client_id: str = oauth_client_id
+
+    def verify_user(self, authorization_header: str) -> User:
+        """Verify the user by checking if the JWT token provided is valid."""
+        jwt_token: str = self._extract_token_from_header(authorization_header)
+        google_certs: Mapping = self._get_google_certs()
+        try:
+            payload: Mapping = jwt.decode(
+                token=jwt_token,
+                certs=google_certs,
+                verify=True,
+                audience=[self.oauth_client_id, "CG-TB-CLIENT"],
```

Comment:
Since this repo itself is the target audience, can we exclude CG here and just have it be "trailblazer"?
```suggestion
                audience=[self.oauth_client_id, "trailblazer"],
```